openpype/modules/deadline/plugins/publish/submit_publish_job.py

```diff
@@ -398,6 +408,13 @@ def process(self, instance):
                 instance, instance_skeleton_data,
                 self.aov_filter, self.skip_integration_repre_list,
                 do_not_add_review)
+            if instance.data.get("expected_families", []):
```

Comment:
I don't understand this logic, what is expected value in `instance.data.get("expectedFiles"instance.data.get("expectedFiles")` ? 
Aren't you re-creating first call of `create_instances_for_aov` again? Shouldn't be enough to add there proper value `farm_cache`, how do you decide it should be actually triggered? 
Arent 'expectedFiles' list all the time? Only for AOVs it contains dictionary, without AOVs just a list of files? (Not sure how does single file look like, but I am expecting ['file1.txt'])

Reply:
1. instance.data.get("expected_families") stores the family type which needs later for file transmission from the work folder (which stores the work cached in deadline) to the publish folder. I can add one more instance.data["farm_cache"] = True to replace the farm_cache == True if it is really
2. I am not quite sure about single file looks like so I follow the logic similar to the image render submission. Maybe I can edit the logic similar to the old implementation of Max deadline (which uses only the list) to see how it goes.

Reply:
Ohh, shh...my bad, got confused by `expectedFiles` and `expected_families`, stupid speedreading course :).

Shouldn't be there:
```
instances = create_instances_for_aov(
                    instance, instance_skeleton_data,
                    self.aov_filter,
                    self.skip_integration_repre_list,
                    do_not_add_review,
                    bool(instance.data.get("expected_families")))
```   
instead of double calls to `create_instances_for_aov`? You are rewriting `instances` from line 407, not extending them, right? (If there are no `expected_families` it will do original instances, if they do, it will create farm cached.)
(BTW all the farm cached families are producing `expectedFiles` as a dict, eg. AOVs, not as a regular list, right?)

Reply:
Expected_files are all in dict and you can find these lines:)
https://github.com/ynput/OpenPype/blob/f963be7cd56db464f6b91c6c0dd6b19785a9c341/openpype/hosts/houdini/plugins/publish/collect_cache_farm.py#L52-L53